lib/modules/hosts/xkcd.js

```diff
@@ -0,0 +1,28 @@
+addLibrary('mediaHosts', 'xkcd', {
+	domains: ['xkcd.com'],
+	logo: 'https://xkcd.com/favicon.ico',
+	detect: href => (/^https?:\/\/(?:www\.|m\.)?xkcd\.com\/([0-9]+)/i).exec(href),
```

Comment:
Ooo, I like this idea. Means media hosts can just provide a regex directly if they ain't doing anything special :)